CMakeLists.txt

```diff
@@ -587,6 +588,111 @@ if (BUILD_TESTING)
             set_property(TEST ${test_target} PROPERTY TIMEOUT 7200)
         endforeach()
     endif()
+
+    if(S2N_FUZZ_TEST)
+        set(SCRIPT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz/runFuzzTest.sh")
+        file(GLOB FUZZ_TEST_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/tests/fuzz/*.c")
+
+        file(GLOB TESTLIB_SRC "tests/testlib/*.c")
+        file(GLOB TESTLIB_HEADERS "tests/testlib/*.h" "tests/s2n_test.h")
+
+        add_library(fuzztest SHARED ${TESTLIB_HEADERS} ${TESTLIB_SRC})
+        target_include_directories(fuzztest PUBLIC tests)
+        target_link_libraries(fuzztest PUBLIC ${PROJECT_NAME})
+
+        # Set default values for fuzzing if not defined
+        if(NOT DEFINED LIBFUZZER_LIB)
+            set(LIBFUZZER_LIB "/usr/local/libfuzzer/lib/libFuzzer.a")
+        endif()
+        
+        if(NOT DEFINED FUZZ_TIMEOUT_SEC)
+            set(FUZZ_TIMEOUT_SEC 120)
+        endif()
+
+        if(NOT DEFINED CORPUS_UPLOAD_LOC)
```

Comment:
I'd prefer to make the fuzz test target a bit "dumber". E.g. I think it should always just assume that the corpus is located in the same place and that the upload location is always the same. After all, how often are we actually configuring those values?

But I think that's gonna be drastically easier to do once we move to cmake idioms (ctest/file locations/etc) so I'm happy to swallow those comments for now 😄